[WPE] WPE Platform: add support for running WebKit API tests with the new API

carlosgcampos opened the pull request:
#### 33804636d79d9b1e9d5a941ae5aa307068343764
<pre>
[WPE] WPE Platform: add support for running WebKit API tests with the new API
<a href="https://bugs.webkit.org/show_bug.cgi?id=293609">https://bugs.webkit.org/show_bug.cgi?id=293609</a>

Reviewed by Alejandro G. Castro.

And switch to use the new API by default when running API tests.

Canonical link: <a href="https://commits.webkit.org/295605@main">https://commits.webkit.org/295605@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9317ea0d0a833277fc08fdb10f66d3d9315ad376

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105624 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25373 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15765 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110817 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/56224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107665 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25834 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33874 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/80213 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/56224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108630 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20180 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95318 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60522 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/19916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13410 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55655 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89629 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/13450 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113664 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32762 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/89288 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33124 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91547 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88952 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33828 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11628 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28207 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17128 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32688 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32434 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35782 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34031 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestInputMethodContext.cpp:
```diff
@@ -108,12 +111,30 @@ static gboolean webkitInputMethodContextMockFilterKeyEvent(WebKitInputMethodCont
-    gunichar character = wpe_key_code_to_unicode(keyval);
-    bool isControl = wpeKeyEvent->modifiers & wpe_input_keyboard_modifier_control;
-    bool isShift = wpeKeyEvent->modifiers & wpe_input_keyboard_modifier_shift;
+    uint32_t keyval;
```
Quite ugly method, gettin' even more ugly. This could need some love & refactoring -- but out of scope for this PR.

nikolaszimmermann commented on Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
```diff
@@ -368,9 +399,15 @@ class Test {
     static Vector<GRefPtr<GDBusConnection>> s_dbusConnections;
     static HashMap<uint64_t, GDBusConnection*> s_dbusConnectionPageMap;
     static WebKitMemoryPressureSettings* s_memoryPressureSettings;
+#if ENABLE(WPE_PLATFORM)
```
I'd merge the two #if sections.

nikolaszimmermann commented on Tools/TestWebKitAPI/glib/WebKitGLib/wpe/WebViewTestWPE.cpp:
```diff
@@ -37,14 +38,32 @@ void WebViewTest::resizeView(int width, int height)
 {
+#if ENABLE(WPE_PLATFORM)
+    if (m_display) {
+        auto* view = webkit_web_view_get_wpe_view(m_webView.get());
```
Here you can assume view not-null?

carlosgcampos replied:
Yes, but we can add an assert anyway.

nikolaszimmermann commented on Tools/TestWebKitAPI/glib/WebKitGLib/wpe/WebViewTestWPE.cpp:
```diff
@@ -37,14 +38,32 @@ void WebViewTest::resizeView(int width, int height)
+#if ENABLE(WPE_PLATFORM)
+    if (m_display) {
+        auto* view = webkit_web_view_get_wpe_view(m_webView.get());
+        if (width && height)
```
Shouldn't we RELEASE_ASSERT this in testing code, to make sure that doesn't happen?

carlosgcampos replied:
No, width and height are optional, so 0 means don't resize. Note that this was written before std::optional, nowadays it would be clearer to use std::optional.

nikolaszimmermann commented on Tools/TestWebKitAPI/glib/WebKitGLib/wpe/WebViewTestWPE.cpp:
```diff
@@ -76,13 +95,56 @@ static unsigned testMouseButtonToWPELegacy(WebViewTest::MouseButton button)
+    case WebViewTest::MouseButton::Secondary:
+        return WPE_BUTTON_SECONDARY;
+    }
+    RELEASE_ASSERT_NOT_REACHED();
```
I always assumed you still need a return value here to please the compiler? return 0?

carlosgcampos replied:
You only need a return with debug-only asserts.

nikolaszimmermann commented on Tools/TestWebKitAPI/glib/WebKitGLib/wpe/WebViewTestWPE.cpp:
```diff
@@ -76,13 +95,56 @@ static unsigned testMouseButtonToWPELegacy(WebViewTest::MouseButton button)
 {
+#if ENABLE(WPE_PLATFORM)
+    if (m_display) {
+        auto modifiers = testModifiersToWPE(mouseModifiers);
```
I'd name this `wpeModifiers` for consistency.

nikolaszimmermann commented on Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
```diff
@@ -67,13 +71,29 @@ PlatformWebView::PlatformWebView(WKPageRef relatedPage)
     WKRelease(m_view);
-    delete m_window;
+#if ENABLE(WPE_PLATFORM)
+    if (g_type_class_peek(WPE_TYPE_DISPLAY))
```
That is sufficient to detect if the _current_ runtime chosen API is the new platform API?

carlosgcampos replied:
Yes.

nikolaszimmermann reviewed (commented):
I checked the code changes, looks fine to me. But someone else has to check the testing infrastructure / Python changes, as I'm not sure if you eventually forgot another important place etc.. (thinking of e.g. `run-benchmark` etc..)

Maybe @clopez needs to check that too.

alexgcastro reviewed (approved):
LGTM!

carlosgcampos committed:
[WPE] WPE Platform: add support for running WebKit API tests with the new API
https://bugs.webkit.org/show_bug.cgi?id=293609

Reviewed by Alejandro G. Castro.

And switch to use the new API by default when running API tests.

Canonical link: https://commits.webkit.org/295605@main
